src/main/java/MyRunnable.java

```diff
@@ -0,0 +1,19 @@
+import org.apache.log4j.Logger;
+
+public class MyRunnable implements Runnable {
+    private static final Logger logger = Logger.getLogger(MyRunnable.class);
+    private Counter counter;
+
+    public MyRunnable(Counter counter) {
+        this.counter = counter;
+    }
+
+    public void run() {
+        synchronized (this.counter) {
```

Comment:
it shoudn't be synchronized, your task is to see how thread race works

Reply:
i think if we use general resources then we will need to use synchronized block